lib/krypto/challenge.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require "rbnacl"
+require "msgpack"
+require "openssl"
+
+module Krypto
+  class Challenge
+    OUTER_CHALLENGE_FIELDS = %i[signature inner].freeze
+    class OuterChallenge < Struct.new(*OUTER_CHALLENGE_FIELDS, keyword_init: true)
+      def to_msgpack(out = "")
+        to_h.to_msgpack(out)
+      end
+    end
+
+    INNER_CHALLENGE_FIELDS = %i[publicEncryptionKey challengeData].freeze
+    class InnerChallenge < Struct.new(*INNER_CHALLENGE_FIELDS, keyword_init: true)
+      def to_msgpack(out = "")
+        to_h.to_msgpack(out)
+      end
+    end
+
+    def self.generate(signing_key, challenge_data)
```

Comment:
Do we need to add a func that accepts a pre-generated key? Or are we good with a new key per challenge?

Reply:
I think we're good with a new one each time